src/routing_node.rs

```diff
@@ -1737,61 +1737,92 @@ impl RoutingNode {
 
         match identity {
             ConnectionName::Routing(routing_name) => {
-                match self.routing_table {
-                    Some(ref mut routing_table) => {
-                        if public_id.name() != routing_name { return false; };
-                        let trigger_churn = routing_table
-                            .address_in_our_close_group_range(&routing_name);
-                        let node_info = NodeInfo::new(public_id,
-                                                      vec![endpoint.clone()],
-                                                      Some(connection));
-                        let routing_table_count_prior = routing_table.size();
-                        let (added, removal_node) = routing_table.add_node(node_info);
-
-                        match removal_node {
-                            Some(node) => {
-                                match node.connection {
-                                    Some(connection) => {
-                                        let _ = self.action_sender.send(
-                                            Action::DropConnections(vec![connection]));
-                                    },
-                                    None => ()
-                                }
-                            },
-                            None => ()
+                if public_id.name() != routing_name {
+                    return false
+                }
+
+                // TODO(Spandan) - Suspicious code - is there a possiblility of getting add_peer()
+                // even if there is no routing table ? When and why would this happen ?
```

Comment:
The routing table should always exist IMHO, I suspect the creation of a routing table was used to promote to a node instead of being a client. So to be clear we should not do this, we should have a routing table if we are a node & only  a node can get an add_peer(). 
To add a peer we make sure the call comes only from a connect_request (if we are receiving this) or a connect_response (if we are receiving this) and we add it if it passes the `routing_table.add()`. IF we think a peer is worth connecting to we do a `routing_table.check()` & if successful then we initiate the connect_request (or connect_response).

Hope that makes sense :-)